src/main/java/Parser.java

```diff
@@ -1,33 +1,123 @@
-import java.io.IOException;
+import java.util.DuplicateFormatFlagsException;
 
 public class Parser {
 
-    public static TaskType returnsTaskType (String command) throws DukeException {
+    public static Command parse (String command) throws DukeException {
 
         String[] words = command.split(" ", 2);
 
+        Task task = null;
+        int index;
 
         switch (words[0]) {
         case "todo":
-            return TaskType.TODO;
+            task = Parser.parsesAdd(command, TaskType.TODO);
+            return new AddCommand(task);
         case "deadline":
-            return TaskType.DEADLINE;
+            task = Parser.parsesAdd(command, TaskType.DEADLINE);
+            return new AddCommand(task);
         case "event":
-            return TaskType.EVENT;
+            task = Parser.parsesAdd(command, TaskType.EVENT);
+            return new AddCommand(task);
         case "done":
-            return TaskType.DONE;
+            index = Parser.parsesDone(command);
+            return new DoneCommand(index);
         case "delete":
-            return TaskType.DELETE;
+            index = Parser.parsesDone(command);
+            return new DeleteCommand(index);
         }
 
         switch (command) {
         case "list":
-            return Bot.TaskType.LIST;
+            return new ListCommand();
         case "bye":
-            return Bot.TaskType.BYE;
+            return new ExitCommand();
         }
 
         throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
 
     }
+
+    public static int parsesDone (String command) throws DukeException {
+
+        String[] words = command.split(" ", 2);
+
+        if (words.length == 1) throw DukeException.emptyDescription();
+
+        if (words[1].isBlank()) throw DukeException.emptyDescription();
+
+        int index = Integer.valueOf(words[1]) - 1;
+
+        return index;
+
+    }
+
+    public static Task parsesAdd (String command, TaskType taskType) throws DukeException {
```

Comment:
Perhaps it may be better to rename your function to an imperative, maybe `parseAdd`? This is so that the function adheres to coding standards, and states "what it will do" rather than "what it does".